plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,342 @@
+        if iso.has_rock_ridge():
+            ISO_MODE = "rr"
+        elif iso.has_joliet():
+            ISO_MODE = "joliet"
+        elif iso.has_udf():
+            ISO_MODE = "udf"
+
+        for op_data in op_data_list:
+            op = op_data['op'].strip()
+            data = op_data['data']
+            if op == "delete_files":
+                for item in data:
+                    ret, msg = iso_delete_file(iso, item.strip())
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+            elif op == "add_files":
+                for item in data:
+                    ret, msg = iso_add_file(iso, item['src_file'],
+                                            item['dest_file'])
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+        iso.write(dest_iso)
+    except Exception as err:
+        msg = "Failed to rebuild ISO %s with error: %s" % (src_iso, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+    finally:
+        if iso:
+            iso.close()
+    return
+
+
+def main():
+    op_data_list = []
+    op_data_item = {}
+    global MODULE_ISO_CUSTOMIZE
+
+    argument_spec = dict(
+        src_iso=dict(type='str', required=True),
+        dest_iso=dict(type='str', required=True),
+        delete_files=dict(type='list', required=False, elements='str'),
+        add_files=dict(type='list', required=False, elements='dict'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+    )
+    if not HAS_PYCDLIB:
+        module.fail_json(
+            missing_required_lib('pycdlib'), exception=PYCDLIB_IMP_ERR)
+
+    src_iso = module.params.get('src_iso')
+    if not os.path.exists(src_iso):
+        module.fail_json(msg="The %s does not exist." % src_iso)
+
+    dest_iso = module.params.get('dest_iso')
+    if not dest_iso:
+        module.fail_json(str(msg='Please specify the path of the customized ISO file \
+            using dest_iso parameter.'))
+
+    dest_iso_dir = os.path.dirname(dest_iso)
+    if dest_iso_dir and not os.path.exists(dest_iso_dir):
+        module.fail_json(msg="The dest directory %s does not exist" % dest_iso_dir)
+
+    MODULE_ISO_CUSTOMIZE = module
+
+    delete_files_list = module.params.get('delete_files')
+    if delete_files_list:
+        op_data_item['op'] = "delete_files"
+        op_data_item['data'] = delete_files_list
+        op_data_list.append(copy.deepcopy(op_data_item))
+
+    add_files_list = module.params.get('add_files')
+    if add_files_list:
+        for item in add_files_list:
+            if not os.path.exists(item['src_file']):
+                module.fail_json(msg="The %s does not exist." % item['src_file'])
+        op_data_item['op'] = "add_files"
+        op_data_item['data'] = add_files_list
+        op_data_list.append(copy.deepcopy(op_data_item))
+
+    iso_rebuild(src_iso, dest_iso, op_data_list)
```

Comment:
Instead of creating this data structure that will only contain "add_files" and "delete_files", wouldn't it be much simpler to pass `add_files_list` and `delete_files_list` directly to `iso_rebuild()`?

Reply:
> Instead of creating this data structure that will only contain "add_files" and "delete_files", wouldn't it be much simpler to pass `add_files_list` and `delete_files_list` directly to `iso_rebuild()`?

Very good practice.  Fix it and thanks very much.